### PR TITLE
SPRNGCORE-4: Add Kafka dependencies

### PR DESCRIPTION
**Pull request by wwelling**
https://folio-org.atlassian.net/browse/SPRNGCORE-4

Testing with mod-worfklow and mod-camunda in progress.

### CODE REVIEW

**Review by Dbreck-TAMU — APPROVED** [review]

File: messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+
+@EnableKafka
+@Configuration
+@Profile("messaging")
+public class KafkaMessageConfig {
+
+  // cut and past from - https://github.com/folio-org/mod-remote-storage/blob/master/src/main/java/org/folio/rs/config/KafkaConfiguration.java
```
Comment (kaladay) [inline messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java:26]:
```suggestion
```
The "Cut and paste" should be documented in the commit.
I see no reason to have that in the code as it clutters it up.

File: messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  @Bean
+  public ConcurrentKafkaListenerContainerFactory<String, Event> kafkaListenerContainerFactory(
+    KafkaProperties kafkaProperties
+  ) {
+    var factory = new ConcurrentKafkaListenerContainerFactory<String, Event>();
+    factory.setBatchListener(true);
+    factory.setConsumerFactory(jsonNodeConsumerFactory(kafkaProperties));
+    // minus lombok @Slf4j
+    // factory.setCommonErrorHandler(new DefaultErrorHandler((exception, data) -> log.error(
+    //  "Error in process with Exception {} and the record is {}", exception, data)));
```
Comment (kaladay) [inline messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java:36]:
```suggestion
```

File: messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    return factory;
+  }
+
+  private ConsumerFactory<String, Event> jsonNodeConsumerFactory(
+    KafkaProperties kafkaProperties
+  ) {
+    var deserializer = new JsonDeserializer<>(Event.class);
```
Comment (kaladay) [inline messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java:44]: We should probably avoid the use of `var` and provide the actualy time.

I'm not sure what but something like:
```
    JsonDeserializer<?> deserializer = new JsonDeserializer<>(Event.class);
```

**Review by kaladay — CHANGES_REQUESTED** [review]

File: messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java (hunk 1 above)
Comment (wwelling) [reply messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java]: Removed.

File: messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java (hunk 3 above)
Comment (wwelling) [reply messaging/src/main/java/org/folio/spring/messaging/config/KafkaMessageConfig.java]: Updated type.

**Review by kaladay — APPROVED** [review]

**Pull request merged**